### PR TITLE
(#931) Update Java Dependency In Jenkins Upgrade Doc

### LINKED ISSUE #931: Update Jenkins Upgrade Doc With New Java Dependency
## What New Or Updated Would You Like To See?

The package name for temuin11jre is misspelled on this [doc page](https://docs.chocolatey.org/en-us/c4b-environments/quick-start-environment/upgrade-jenkins).

While i'm at will update the dependency to JRE21

## Why Is It Needed?
Correct spelling mistake in documentation. Also bring up to date.

## Additional Context?
* [ZenDesk Ticket](https://chocolatey.zendesk.com/agent/tickets/20824)

### PR DESCRIPTION
**Pull request by ryanrichter94**
## Description Of Changes
- Update dependency to point out temurin21jre pkg
- Add link to Jenkins own Java support policy docs
- Fix typos and incorrect package references

## Motivation and Context
Documentation needed updated to reflect Jenkins own guidance for java dependency

## Testing
* [X] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made
* [X] Minor documentation fix (typos etc.).
* [X] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* ~[ ] New documentation page added.~
* ~[ ] The change I have made should have a video added, and I have raised an issue for this.~
    ~* Issue #~

## Change Checklist
* ~[ ] Requires a change to menu structure (top or left-hand side)/~
* ~[ ] Menu structure has been updated~

## Related Issue
Fixes #931

### CODE REVIEW

**Review by steviecoaster — DISMISSED** [review]

**Review by steviecoaster — CHANGES_REQUESTED** [review]

File: input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md [hunk 1]
```diff
@@ -13,20 +13,26 @@ If your server is restricted then internalize the package to your internal repos
 
 > :choco-warning: **WARNING**
 >
-> The current Jenkins package requires a Java 18 or 11 which hasn't been added as a dependency (due to the numerous flavours of Java out there). As part of the Quick Start Guide setup we install the temurin11jre package. However any Java version 11 or 18 package will work.
+> The current Jenkins package requires Java version 17 or 21 which hasn't been added as a package dependency to jenkins (due to the numerous flavours of Java out there). As part of the Quick Start Guide setup we install the temurin21jre package. However any Java version 17 or 21 package will work.
+>
+> Here is a link to Jenkins own [java support policy documentation](https://www.jenkins.io/doc/book/platform-information/support-policy-java/).
```
[the remark this answers is not in the document]
Comment (ryanrichter94) [reply input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md:18]: fixed

File: input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md [hunk 2]
```diff
@@ -13,20 +13,26 @@ If your server is restricted then internalize the package to your internal repos
...
 
 ## Instructions
 
-1. Internalize the nexus-repository package and push to your internal repo
-2. Internalize a java package compatible with Jenkins and push to your internal repo. We recommend the [temurin11jre package](https://community.chocolatey.org/packages/Temurin11jre) used during setup in the Quick Start Guide.
-3. choco upgrade the temurin11jre and nexus-repository packages (Example command provided below)
+1. Internalize the Jenkins package and push to your internal repo
+2. Internalize a java package compatible with Jenkins and push to your internal repo. We recommend the [temurin21jre package](https://community.chocolatey.org/packages/Temurin21jre).
+3. choco upgrade the temurin21jre and Jenkins packages (Example command provided below)
```
[the remark this answers is not in the document]
Comment (ryanrichter94) [reply input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md:24]: fixed

File: input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md [hunk 3]
```diff
@@ -13,20 +13,26 @@ If your server is restricted then internalize the package to your internal repos
...
+> The current Jenkins package requires Java version 17 or 21 which hasn't been added as a package dependency to jenkins (due to the numerous flavours of Java out there). As part of the Quick Start Guide setup we install the temurin21jre package. However any Java version 17 or 21 package will work.
+>
+> Here is a link to Jenkins own [java support policy documentation](https://www.jenkins.io/doc/book/platform-information/support-policy-java/).
 
 ## Instructions
 
-1. Internalize the nexus-repository package and push to your internal repo
-2. Internalize a java package compatible with Jenkins and push to your internal repo. We recommend the [temurin11jre package](https://community.chocolatey.org/packages/Temurin11jre) used during setup in the Quick Start Guide.
-3. choco upgrade the temurin11jre and nexus-repository packages (Example command provided below)
+1. Internalize the Jenkins package and push to your internal repo
```
[the remark this answers is not in the document]
Comment (ryanrichter94) [reply input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md:22]: fixed

**Comment by steviecoaster** [comment]
This section needs changed:

```
This document outlines the process for upgrading Jenkins running inside our Quick Start Environment.
The script provided assumes your server has access to the internet to download the Jenkins package from the community repository.
If your server is restricted then internalize the package to your internal repository and update the source in the script appropriately.
```

There is no script referenced on this page anywhere, and where there _is_ code, we have placeholders for an internal repo url.

Instructions on this page call our internalization, so I think it's just this section that needs tweaked.

The language should _never_ assume pulling from the Chocolatey Community Repository, but rathe the internalization of the package, and then installation from an internal source, even in these seemingly one-off cases.

As we are updating documentation be sure to keep an eye out for this sort of thing, and change the language in the doc when found.

**Review by steviecoaster — CHANGES_REQUESTED** [review]
See previous comment. it won't let me suggest changes on the lines that need it for some reason.

**Comment by ryanrichter94** [comment]
> This section needs changed:
> 
> ```
> This document outlines the process for upgrading Jenkins running inside our Quick Start Environment.
> The script provided assumes your server has access to the internet to download the Jenkins package from the community repository.
> If your server is restricted then internalize the package to your internal repository and update the source in the script appropriately.
> ```
> 
> There is no script referenced on this page anywhere, and where there _is_ code, we have placeholders for an internal repo url.
> 
> Instructions on this page call out internalization, so I think it's just this section that needs tweaked.
> 
> The language should _never_ assume pulling from the Chocolatey Community Repository, but rather the internalization of the package, and then installation from an internal source, even in these seemingly one-off cases.
> 
> As we are updating documentation be sure to keep an eye out for this sort of thing, and change the language in the doc when found.

Updated wording used in this section above with last commit.

File: input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md [hunk 4]
```diff
@@ -7,26 +7,32 @@ Description: How to upgrade jenkins
 
 ## Upgrade Jenkins in Quick Start Environment
 
-This document outlines the process for upgrading Jenkins running inside our Quick Start Environment.
-The script provided assumes your server has access to the internet to download the Jenkins package from the community repository.
-If your server is restricted then internalize the package to your internal repository and update the source in the script appropriately.
+This document outlines the process for upgrading Jenkins, and it's Java dependency running on your Quick Start Environment.
+This guide assumes your server has access to the internet to internalize the needed packages from the community repository.
```
Comment (steviecoaster) [inline input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md:11]:
```suggestion
Chocolatey community repository
```

File: input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md [hunk 5]
```diff
@@ -7,26 +7,32 @@ Description: How to upgrade jenkins
...
+If your server is internet restricted, please internalize the needed packages on a machine that does have internet access. Then push these packages to your internal repository.
```
Comment (steviecoaster) [inline input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md:12]:
```suggstion
If your server is internet restricted, please internalize the needed packages on a machine that does have internet access, then push these packages to your internal repository.
```

**Review by steviecoaster — CHANGES_REQUESTED** [review]

File: input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md (hunk 5 above)
Comment (ryanrichter94) [reply input/en-us/c4b-environments/quick-start-environment/upgrade-jenkins.md]: fixed

**Review by steviecoaster — APPROVED** [review]

**Pull request merged**